src/org/jetbrains/plugins/scala/codeInsight/intention/IntentionUtil.scala

```diff
@@ -0,0 +1,63 @@
+package org.jetbrains.plugins.scala.codeInsight.intention
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.impl.source.codeStyle.CodeEditUtil
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.{PsiWhiteSpace, PsiComment, PsiElement}
+import org.jetbrains.plugins.scala.extensions.PsiElementExt
+
+import scala.collection.mutable.ArrayBuffer
+
+object IntentionUtil {
+  def collectComments(element: PsiElement, onElementLine: Boolean = false): CommentsAroundElement = {
+    def hasLineBreaks(whiteSpace: PsiElement): Boolean = {
+      if (!onElementLine) false
+      else StringUtil.containsLineBreak(whiteSpace.getText)
+    }
+
+    def getElements(newElement: (PsiElement) => Option[PsiElement]): Seq[PsiElement] = {
```

Comment:
It's definitely tailrec function. Just immutable collection instead of ArrayBuffer can be used.